runner/custom_template_factory.py

```diff
@@ -62,6 +62,66 @@ def set_custom_image(in_memory_json_object: str, VM_image_URL: str, VM_image_typ
     # TODO For handling windows and centos the nodeAgentSKUId needs to be changed in the json send object
     # The centos image hasn't been tested so I haven't written logic for handling the VM_image_type
 
+def set_pool_resource_file(in_memory_json_object: str, resource_branch_name: str):
+    """
+    Replaces the github branch name to point to a new branch. 
+    "https://raw.../master/..../install-azure-nc-drivers.cmd",
+    "https://raw.../'resource_branch_name'/..../install-azure-nc-drivers.cmd",
+
+    This is for testing out new ps1 or bat files
+
+    :param in_memory_json_object: The json object that needs to be updated with a repo file path
+    :type in_memory_json_object: str
+    :param resource_branch_name: The value that needs to be set
+    :type resource_branch_name: str
+
+    """
+    if not resource_branch_name:
+        return
+
+    try:
+        if in_memory_json_object["pool"]["startTask"]["resourceFiles"] is not None:
+            resource_files = in_memory_json_object["pool"]["startTask"]["resourceFiles"]
+            for index in range(len(resource_files)):
+                resource_file = resource_files[index]
+                for key in resource_file:
+                    resource_file[key] = resource_file[key].replace("batch-extension-templates/master", "batch-extension-templates/"+resource_branch_name.strip())
+    except KeyError:
+        pass
+    except AttributeError: 
+        pass
+
+
+def set_job_resource_file(in_memory_json_object: str, resource_branch_name: str):
```

Comment:
These two methods are pretty much the same.  You could have a common method 'update_resource_file_url(resource_files, branch_name)' that has the looping/replacement logic.